src/components/AugmentedResultList/AugmentedResultList.ts

```diff
@@ -0,0 +1,142 @@
+
+/**
+ * Interface for the object data returned from remote action.
+ *
+ * @export
+ * @interface IObjectData
+ */
+export interface IObjectData {
+    /**
+     * Data specific to a result with matching object id.
+     */
+    objectData: {}[];
+    /**
+     * Data to add to every result with matching object id.
+     */
+    commonData: {};
+}
+
+/**
+ * Generic interface for the response returned by the remote action method.
+ *
+ * @export
+ * @interface IPromiseReturnArgs
+ * @template T
+ */
+export interface IPromiseReturnArgs {
+    data: IObjectData;
+}
+
+export interface AugmentedResultListOptions extends IResultListOptions {
+    matchingIdField: IFieldOption;
+    objectDataAction: (objectIds: String[]) => Promise<IPromiseReturnArgs>;
+}
+
+export class AugmentedResultList extends Coveo.ResultList implements IComponentBindings {
+    static ID = 'AugmentedResultList';
+
+    /**
+     * @componentOptions
+     */
+    static options: AugmentedResultListOptions = {
+        /**
+         * The field to be used as matching ID between object and result.
+         */
+        matchingIdField: ComponentOptions.buildFieldOption({
+            required: true,
+        }),
+        /**
+         * The function used to fetch extra object information.
+         */
+        objectDataAction: ComponentOptions.buildCustomOption<(objectIds: String[]) => Promise<IPromiseReturnArgs>>(() => {
+            return null;
+        }),
+    };
+
+    /**
+     * Creates a new `AugmentedResultList` component.
+     * @param element The HTMLElement on which to instantiate the component.
+     * @param options The options for the `ResultList` component.
+     * @param bindings The bindings that the component requires to function normally. If not set, these will be
+     * automatically resolved (with a slower execution time).
+     */
+    constructor(public element: HTMLElement, public options: AugmentedResultListOptions, public bindings: IComponentBindings) {
+        super(element, ComponentOptions.initComponentOptions(element, AugmentedResultList, options), bindings, AugmentedResultList.ID);
+    }
+
+    private getObjectPayload(results: IQueryResult[]): String[] {
+        const field = this.getFieldString(this.options.matchingIdField);
+        if (results.length > 0) {
+            return results.filter((result) => result.raw && result.raw[field]).map((result) => result.raw[field]);
+        }
+        return [];
+    }
+
+    private getFieldString(fieldName: IFieldOption) {
+        return fieldName.replace('@', '');
+    }
+
+    protected enableAnimation() {
+        if (!document.getElementById('overlay')) {
+            $$(document.body).append($$('div', { id: 'overlay', class: 'modal-backdrop fade in' }).el);
+        }
+    }
+
+    protected disableAnimation() {
+        if (document.getElementById('overlay')) {
+            document.getElementById('overlay').remove();
+        }
```

Comment:
How about a oneliner?
```suggestion
        document.getElementById('overlay')?.remove();
```
(Yep, it works after a function call was not sure either, see [TS Playground here](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAtgTwGIFczCgS3DAvDACgCcBTKFIsAZTigAsBzALhgCMQQAbEgQzAEo8APhilylGvQYwA-DADeMAA5EQSlgX64hAchB0EPHTAC+MFmgAmJAGaYwJSwG4AUKEhcSAOk4gGBRFR0LHACGx5OCBJ+GS8VNU1+IA))

Reply:
Haha now i'm embarrassed. I took this from the class PS made without thinking of changing it citing: "if it ain't broke, don't fix it". Should've fixed it :P